dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -66,265 +69,189 @@ public static void prepare() throws Exception {
-            String uri = "/test1_" + i;
-            String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
-            int action = 200;
-            int order = 1;
-
-            contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
-                    forwardTo, action, order, defaultLanguageId);
-
-            VanityUrl vanity1 = vanityUrlAPI.getVanityUrlFromContentlet(contentletAPI.find(contentlet1.getInode(), user, false));
-
-            Assert.assertNotNull(vanity1);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanity1.getInode());
-            Assert.assertEquals(title, vanity1.getTitle());
-            Assert.assertEquals(site, vanity1.getSite());
-            Assert.assertEquals(uri, vanity1.getURI());
-            Assert.assertEquals(forwardTo, vanity1.getForwardTo());
-            Assert.assertEquals(action, vanity1.getAction());
-
-            CachedVanityUrl vanityCached = vanityUrlAPI.getLiveCachedVanityUrl(uri, defaultHost, defaultLanguageId, user);
-
-            Assert.assertEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanityCached.getVanityUrlId());
-
-        } catch (DotDataException | DotSecurityException e) {
-            e.printStackTrace();
-            fail();
-        } finally {
-            try {
-                if (contentlet1 != null) {
-                    contentletAPI.destroy(contentlet1, user, false );
-                }
-            } catch (Exception e) {
-                Logger.error(this.getClass(), "Error cleaning up Vanity Url Links");
-            }
-        }
-    }
+        long i = System.currentTimeMillis();
+        String title = "VanityURL" + i;
+        String site = defaultHost.getIdentifier();
+        String uri = "/test1_" + i;
+        String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
+        int action = 200;
+        int order = 1;
+
+        Contentlet contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
+
+        VanityUrl vanity1 = vanityUrlAPI.fromContentlet(contentletAPI.find(contentlet1.getInode(), user, false));
 
+        Assert.assertNotNull(vanity1);
+        Assert.assertEquals(title, vanity1.getTitle());
+        Assert.assertEquals(site, vanity1.getSite());
+        Assert.assertEquals(uri, vanity1.getURI());
+        Assert.assertEquals(forwardTo, vanity1.getForwardTo());
+        Assert.assertEquals(action, vanity1.getAction());
+        Assert.assertEquals(order, vanity1.getOrder());
+
+    }
+    
+    
+    
+    
     /**
-     * Creates a couple of vanity urls, publish only the first one.
-     * Checks that the first one is in cache and the second one is not.
+     * Creates a vanity url but no publish it, gets it and checks there is not in cache.
      *
      * @see VanityUrlAPI
      */
     @Test
-    public void getLiveVanityUrl() {
+    public void working_vanities_do_not_resolve() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)